### ISSUE TITLE
Require macOS Catalina or later

### ISSUE THREAD

**Issue by low-batt** [iina/iina#4837]
**What you want IINA to do:**
Require the Mac to be running [macOS Catalina](https://en.wikipedia.org/wiki/MacOS_Catalina) (version 10.15) or later.

**What IINA does currently:**
IINA is currently documented as requiring the Mac to be running [OS X El Capitan](https://en.wikipedia.org/wiki/OS_X_El_Capitan) (version 10.11) or later. As discussed in issue #4634, IINA 1.3.2 through 1.3.4 actually require [macOS High Sierra](https://en.wikipedia.org/wiki/MacOS_High_Sierra) (version 10.13) or later. See that issue for further discussion regarding support for [OS X El Capitan](https://en.wikipedia.org/wiki/OS_X_El_Capitan) and [macOS Sierra](https://en.wikipedia.org/wiki/MacOS_10.12).

**Why you think this should be added:**
IINA is built on top of source code (libraries and frameworks) from other projects. This code can be seen if you look inside the IINA application:
<details>
<summary><b>Frameworks:</b></summary>

```text
low-batt@gag ~$ ls /Applications/IINA.app/Contents/Frameworks/
Sparkle.framework		libharfbuzz.0.dylib		libsoxr.0.dylib			libswiftos.dylib
libX11.6.dylib			libhogweed.6.dylib		libspeex.1.dylib		libswiftsimd.dylib
libXau.6.dylib			libidn2.0.dylib			libswiftAVFoundation.dylib	libswresample.4.dylib
libXdmcp.6.dylib		libintl.8.dylib			libswiftAppKit.dylib		libswscale.7.dylib
libarchive.13.dylib		libjpeg.8.dylib			libswiftCore.dylib		libtasn1.6.dylib
libass.9.dylib			liblcms2.2.dylib		libswiftCoreAudio.dylib		libuchardet.0.dylib
libavcodec.60.dylib		libluajit-5.1.2.dylib		libswiftCoreData.dylib		libunibreak.5.dylib
libavdevice.60.dylib		liblz4.1.dylib			libswiftCoreFoundation.dylib	libunistring.5.dylib
libavfilter.9.dylib		liblzma.5.dylib			libswiftCoreGraphics.dylib	libvidstab.1.2.dylib
libavformat.60.dylib		libmpv.2.dylib			libswiftCoreImage.dylib		libwebp.7.dylib
libavutil.58.dylib		libmujs.dylib			libswiftCoreMedia.dylib		libwebpmux.3.dylib
libb2.1.dylib			libnettle.8.dylib		libswiftDarwin.dylib		libxcb-shape.0.dylib
libbluray.2.dylib		libp11-kit.0.dylib		libswiftDispatch.dylib		libxcb-shm.0.dylib
libdav1d.6.dylib		libpcre2-8.0.dylib		libswiftFoundation.dylib	libxcb-xfixes.0.dylib
libfontconfig.1.dylib		libpng16.16.dylib		libswiftIOKit.dylib		libxcb.1.dylib
libfreetype.6.dylib		libpostproc.57.dylib		libswiftMetal.dylib		libzimg.2.dylib
libfribidi.0.dylib		librubberband.2.dylib		libswiftNetwork.dylib		libzmq.5.dylib
libglib-2.0.0.dylib		libsamplerate.0.dylib		libswiftObjectiveC.dylib	libzstd.1.dylib
libgmp.10.dylib			libsharpyuv.0.dylib		libswiftQuartzCore.dylib
libgnutls.30.dylib		libsnappy.1.dylib		libswiftSafariServices.dylib
libgraphite2.3.dylib		libsodium.23.dylib		libswiftXPC.dylib
low-batt@gag ~$ 
```
</details>

This means that IINA can only support the macOS versions supported by all of the code obtained from other projects.

For audio/video and subtitle rendering support IINA uses a library from the [mpv](https://github.com/mpv-player/mpv) project. This is IINA's primary dependency.

From the [System requirements](https://github.com/mpv-player/mpv?tab=readme-ov-file#system-requirements) section of mpv's README:
> A not too ancient Linux (usually, only the latest releases of distributions are actively supported), Windows 10 or later, or macOS 10.15 or later.

The [mpv](https://github.com/mpv-player/mpv) project has raised their minimum supported macOS version to [macOS Catalina](https://en.wikipedia.org/wiki/MacOS_Catalina). Code that supported older versions of macOS has been removed from the source code.

IINA has no choice but to also raise its minimum supported macOS version to match. :disappointed:

**Comment by MrPhil17** [comment]
Unluckily I can confirm. I tried on my old 17" macBook Pro mid2007, the latest version for El Capitan is 1.3.1. When I tried to start versions from 1.3.2, they required High Sierra 10.13 or above.

Little edit: 1.3.1 keep crashing on my MBP when I try to open a video (basic 1080P or 4K MP4 recorded with my phone), I had to downgrade to 1.3.0 for a good experience.

**Comment by saagarjha** [comment]
Seems like this is the reason for the bump: https://github.com/haasn/libplacebo/issues/199. Perhaps it's something we can step in and fix?

**Comment by low-batt** [comment]
@saagarjha [This comment](https://github.com/mpv-player/mpv/commit/23de1deaaaace05bb9504567a852ac66e76a1ad1#commitcomment-135659306) lists some of the reasons why `mpv` raised the minimum macOS version supported. As you can see from that commit they have been going through the sources and stripping out code that supports the older versions. I don't think we can convince that project to reverse course at this point.

@MrPhil17 I avoided saying much about the situation with support for [OS X El Capitan](https://en.wikipedia.org/wiki/OS_X_El_Capitan) as such discussions should occur in issue #4634. Does the 1.3.1 crash match the crash reported in issue #4062? Issue #3327 is another known 10.11 crash. That one has a workaround.

The current plan is for the next release of IINA to be a bug fix release that does not involve upgrading dependences. Restoring support for 10.11 & 10.12 in that release is being investigated. Not clear if that effort will be successful.

After that release IINA will need to upgrade dependencies. When that happens IINA will need to address this issue.

**Comment by MrPhil17** [comment]
> Does the 1.3.1 crash match the crash reported in issue #4062?


This is my [crash report](https://paste.ofcode.org/Detq5LTKGtkYqNCVQhhpUh). Seems the same issue.

**Comment by low-batt** [comment]
Yes, I looked at the crash report and confirmed that is the problem in issue #4062. A rather disturbing problem as that code should not be sensitive to the macOS version.

**Comment by low-batt** [comment]
The Xcode project has been updated to build for 10.15+ in commit 2d013fe6b044298a581bbcbd77420ca3366267aa which has been merged into the `develop` branch.

**Comment by low-batt** [comment]
IINA 1.4.0-beta1 requires macOS Catalina or later.
